types/react-native/index.d.ts

```diff
@@ -3428,22 +3428,22 @@ export interface ViewabilityConfig {
 /**
  * @see https://facebook.github.io/react-native/docs/flatlist.html#props
  */
-export interface FlatListProperties<ItemT> {
+export interface FlatListProperties<ItemT> extends ScrollViewProperties {
 
     /**
      * Rendered in between each item, but not at the top or bottom
      */
     ItemSeparatorComponent?: React.ComponentClass<any> | null
 
     /**
-     * Rendered at the bottom of all the items.
+     * Rendered at the very end of the list.
      */
-    ListFooterComponent?: React.ComponentClass<any> | null
+    ListFooterComponent?: React.ComponentClass<any> | (() => React.ReactElement<any>) | null
```

Comment:
Btw, were you unable to use `React.StatelessComponent<any>`, as [suggested](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/16820#issuecomment-304944375)?